lib/records/P_RecCore.cc

```diff
@@ -445,7 +443,7 @@ RecSetRecord(RecT rec_type, const char *name, RecDataT data_type, RecData *data,
     } else {
       err = send_set_message(r1);
     }
-    g_records_ht.emplace(name, r1);
+    g_records_ht.emplace(r1->name, r1);
```

Comment:
It's very important to use `r1->name` and not `name` because it's a raw pointer, and a pointer to `name` will go stale, but not a pointer to `r1->name` (during the lifetime of `r1`). I suspect some of the earlier reported problems with using `string_view` as a key stem from something like that.